wp-admin/js/widgets/media-widgets.js

```diff
@@ -713,6 +713,20 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.attachment_id = mediaFrameProps.id;
 			}
 
+			isEmbed = 0 === modelProps.attachment_id;
+
+			// Reset all video/audio sources.
+			wp.media.view.settings.embedExts.map( function( ext ) {
+
+				// Only reset when the main source changes.
+				if ( ! ( ext in modelProps ) || isEmbed ) {
+					modelProps[ ext ] = '';
+				}
+			} );
+
+			extension = mediaFrameProps.url.split( '.' ).pop();
+			modelProps[ extension ] = mediaFrameProps.url;
```

Comment:
If we do it for uploaded files as well, we'll automatically populate the additional sources part in the edit modal.

I'm not sure I understand why you want to move away from uploaded files using `url`. Should that be set no matter what? When would we encounter an attachment without it?